queue_job/migrations/12.0.2.3.0/post-migration.py

```diff
@@ -0,0 +1,47 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        _logger.info("Computing exception name for failed jobs")
+        _compute_jobs_new_values(env)
+
+
+def _compute_jobs_new_values(env):
+    for job in env["queue.job"].search(
+        [("state", "=", "failed"), ("exc_info", "!=", False)]
+    ):
+        exception_details = _get_exception_details(job)
+        if exception_details:
+            job.update(exception_details)
+
+
+def _get_exception_details(job):
+    for line in reversed(job.exc_info.splitlines()):
+        if _find_exception(line):
+            name, msg = line.split(":", 1)
+            return {
+                "exc_name": name.strip(),
+                "exc_message": msg.strip("()', \""),
+            }
```

Comment:
Better one reading all the exc_info at once:
```suggestion
def _compute_jobs_new_values(env):
    job_exc_infos = env["queue.job"].search(
        [("state", "=", "failed"), ("exc_info", "!=", False)]
    ).read(fields=["exc_info"])
    for job_row in job_exc_infos:
        exception_details = _get_exception_details(job_row["exc_info"])
        if exception_details:
            job = env["queue.job"].browse(job_row["id"])
            job.write(exception_details)


def _get_exception_details(exc_info):
    for line in reversed(exc_info.splitlines()):
        if _find_exception(line):
            name, msg = line.split(":", 1)
            return {
                "exc_name": name.strip(),
                "exc_message": msg.strip("()', \""),
            }
```

Reply:
Hi @etobella, could you merge this suggestion?

Reply:
Sure!!! Done!

Reply:
Thanks :)